components-api/src/main/scala/pl/touk/nussknacker/engine/api/typed/FromInstanceTypeDeterminer.scala

```diff
@@ -35,15 +42,25 @@ object FromInstanceTypeDeterminer {
       case other =>
         Typed(other.getClass) match {
           case typedClass: TypedClass =>
+            val adjustedType = replaceWithGenericTypeIfNeeded(typedClass)
             ToJsonEncoder.default.encode(other) match {
-              case Valid(_)   => TypedObjectWithValue(typedClass, other)
-              case Invalid(_) => typedClass
+              case Valid(_)   => TypedObjectWithValue(adjustedType, other)
+              case Invalid(_) => adjustedType
             }
           case notTypedClass => notTypedClass
         }
     }
   }
 
+  // We don't want to present very specific types such as ZoneRegion or sun.nio.cs.UTF_8
+  private def replaceWithGenericTypeIfNeeded(typedClass: TypedClass) = {
+    if (classOf[ZoneId].isAssignableFrom(typedClass.klass) && !classOf[ZoneOffset].isAssignableFrom(typedClass.klass))
+      Typed.typedClass[ZoneId]
+    else if (classOf[Charset].isAssignableFrom(typedClass.klass))
+      Typed.typedClass[Charset]
+    else typedClass
```

Comment:
could be more readable with pattern match
```suggestion
    typedClass.klass match {
      case c if classOf[ZoneId].isAssignableFrom(c) && !classOf[ZoneOffset].isAssignableFrom(c) =>
        Typed.typedClass(classOf[ZoneId])
      case c if classOf[Charset].isAssignableFrom(c) =>
        Typed.typedClass(classOf[Charset])
      case _ =>
        typedClass
    }
```